From galak@kernel.crashing.org  Thu Jul  3 15:35:41 2008
From: Kumar Gala <galak@kernel.crashing.org>
Date: Wed, 2 Jul 2008 02:14:33 -0500 (CDT)
Subject: USB: EHCI: Reconciling USB register differences on MPC85xx vs MPC83xx
To: dbrownell@users.sourceforge.net, Greg KH <greg@kroah.com>
Cc: linux-usb@vger.kernel.org, linuxppc-dev@ozlabs.org
Message-ID: <Pine.LNX.4.64.0807020213250.25180@blarg.am.freescale.net>


From: Srikanth Srinivasan <srikanth.srinivasan@freescale.com>

A couple of USB register initializations had to be changed on MPC85xx
platforms.  This is due to the internal SoC buses being different on
MPC83xx SoCs vs MPC85xx SoCs.

We currently handle this via an ifdef since 83xx and 85xx are mutually
exclusive kernel builds.

Signed-off-by: Srikanth Srinivasan <srikanth.srinivasan@freescale.com>
Signed-off-by: Kumar Gala <galak@kernel.crashing.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/ehci-fsl.c |    5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/usb/host/ehci-fsl.c
+++ b/drivers/usb/host/ehci-fsl.c
@@ -230,8 +230,13 @@ static void mpc83xx_usb_setup(struct usb
 
 	/* put controller in host mode. */
 	ehci_writel(ehci, 0x00000003, non_ehci + FSL_SOC_USB_USBMODE);
+#ifdef CONFIG_PPC_85xx
+	out_be32(non_ehci + FSL_SOC_USB_PRICTRL, 0x00000008);
+	out_be32(non_ehci + FSL_SOC_USB_AGECNTTHRSH, 0x00000080);
+#else
 	out_be32(non_ehci + FSL_SOC_USB_PRICTRL, 0x0000000c);
 	out_be32(non_ehci + FSL_SOC_USB_AGECNTTHRSH, 0x00000040);
+#endif
 	out_be32(non_ehci + FSL_SOC_USB_SICTRL, 0x00000001);
 }
 
